### PR TITLE
Revert "Updating build farm cluster-groups"

### PR DESCRIPTION
**Pull request by hongkailiu**
Reverts openshift/release#33062

A workaround has been applied to b02.
Let us move jobs back.

/cc @openshift/test-platform @bradmwilliams

### CODE REVIEW

File: hack/generators/release-controllers/content/osd_rc_deployments.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def _add_osd_rc_deployment(gendoc):
 
     # Creating Cluster Groups for the AMD64 jobs...
     if context.arch == 'x86_64':
-        extra_rc_args.append('--cluster-group=build01,build03,build04,build05')
+        extra_rc_args.append('--cluster-group=build02,build03,build04')
```
Comment (bradmwilliams) [inline hack/generators/release-controllers/content/osd_rc_deployments.py:229]: We probably dont want to revert this change completely, but simply add back `build02`.

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: hack/generators/release-controllers/content/osd_rc_deployments.py (hunk 1 above)
Comment (hongkailiu) [reply hack/generators/release-controllers/content/osd_rc_deployments.py:229]: Ack. Let me revise.

Comment (hongkailiu) [reply hack/generators/release-controllers/content/osd_rc_deployments.py]: Done!

**Review by bradmwilliams — APPROVED** [review]
